### ISSUE TITLE
Where is "the" source code

### ISSUE THREAD

**Issue by devbuzzer** [IntelliTect/EssentialCSharp#572]
Hi, the first thing I do when I wish to buy a book is see the source code.
As far as the samples I got from the book preview, there are blocks of code that are abbreviated with ellipsis (...) requiring the reader to look at the source code.
But, where is the real source code for me in order to match watch I typed when reading the book?
As far as I can see, this repo is for a solution that generates the output of the source code, and not "the" source code.
How I am supposed to read a book full of (...) that the source code is obfuscated?
Am I supposed to "watch" what the author is able to do? Not read?

**Comment by BenjaminMichaelis** [comment]
Hello @devbuzzer, thank you for this feedback! 

Is there specific case that you are referring to? 

All the source code in this repository does build and run, and most have associated tests alongside so it is exactly what the author is able to do. Some of the cases where it shows ellipses in the book, there is more source code in this repository for that obfuscated section, but on the occasions there isn't, it is for elucidation reasons but the code still compiles and runs.

**Comment by devbuzzer** [comment]
Yes, the code builds and run, but it's not the source code from the book.
There are a lot of:

#region INCLUDE
#endregion INCLUDE

**Comment by BenjaminMichaelis** [comment]
> Yes, the code builds and run, but it's not the source code from the book. There are a lot of:
> 
> #region INCLUDE #endregion INCLUDE

I will work on adding some documentation in the readme about those. Basically what you can see in the book is between the "#region INCLUDE" and the "#endregion INCLUDE" minus whatever is between a "#region EXCLUDE" and a "#endregion EXCLUDE". Also the source code is more current than the latest print edition on the main v12.0 branch. If you are looking at the most recent print book, the source code should match what is here: https://github.com/IntelliTect/EssentialCSharp/tree/v8.0-release . But if you are looking at the online version, it should be up to date with the v12.0 branch.

**Comment by devbuzzer** [comment]
I had plans to buy the new book Essential C# 12.0, 8th Edition from Oreily https://www.oreilly.com/library/view/essential-c-120/9780138219550/ after its release. But I dislike the way the source code has been shared. It makes no sense!

**Comment by BenjaminMichaelis** [comment]
> I had plans to buy the new book Essential C# 12.0, 8th Edition from Oreily https://www.oreilly.com/library/view/essential-c-120/9780138219550/ after its release. But I dislike the way the source code has been shared. It makes no sense!
> 
> 

Is there something that we can do to make it more clear?
